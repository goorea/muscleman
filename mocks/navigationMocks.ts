export const navigationDispatchMock = jest.fn();
export const navigationNavigateMock = jest.fn();
export const navigationGoBackMock = jest.fn();
export const navigationSetOptionsMock = jest.fn();
export const navigationSetParamsMock = jest.fn();
export const navigationCanGoBackMock = jest.fn();
export const navigationGetParentMock = jest.fn();
export const navigationGetStateMock = jest.fn();
export const navigationReplaceMock = jest.fn();
export const navigationIsFocusedMock = jest.fn();
export const navigationResetMock = jest.fn();
export const navigationPopMock = jest.fn();
export const navigationPushMock = jest.fn();
export const navigationPopToTopMock = jest.fn();
export const navigationAddListenerMock = jest.fn();
export const navigationRemoveListenerMock = jest.fn();
export const navigationJumpToMock = jest.fn();
export const navigationMock = {
  dispatch: navigationDispatchMock,
  navigate: navigationNavigateMock,
  goBack: navigationGoBackMock,
  setOptions: navigationSetOptionsMock,
  setParams: navigationSetParamsMock,
  canGoBack: navigationCanGoBackMock,
  getParent: navigationGetParentMock,
  getState: navigationGetStateMock,
  isFocused: navigationIsFocusedMock,
  reset: navigationResetMock,
  addListener: navigationAddListenerMock,
  removeListener: navigationRemoveListenerMock,
  replace: navigationReplaceMock,
  pop: navigationPopMock,
  push: navigationPushMock,
  popToTop: navigationPopToTopMock,
  jumpTo: navigationJumpToMock,
};
