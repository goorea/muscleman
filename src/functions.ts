import { Source } from 'react-native-fast-image';
import StackFlashMessage, { Options } from 'react-native-stack-flash-message';

import { WeightSet } from '@src/types';
import { Set, TrainingType, User } from '@src/types/graphql';

export const flash = (options: Options) => StackFlashMessage.show(options);

export const getTrainingTypeForKorean = (type: TrainingType): string => {
  switch (type) {
    case TrainingType.Lower:
      return '하체';
    case TrainingType.Chest:
      return '가슴';
    case TrainingType.Back:
      return '등';
    case TrainingType.Shoulder:
      return '어깨';
    case TrainingType.Arm:
      return '팔';
    case TrainingType.Abdominal:
      return '복근';
    case TrainingType.Cardiovascular:
      return '유산소';
    default:
      return '기타';
  }
};

export const isWeightSets = (sets: Set[]): sets is WeightSet[] =>
  sets.every(
    set =>
      (set as WeightSet).weight !== undefined &&
      (set as WeightSet).count !== undefined,
  );

export const getProfileImage = (user: User): Source | number =>
  user.profileImagePath
    ? { uri: user.profileImagePath }
    : require('@src/resources/images/mock.png');
