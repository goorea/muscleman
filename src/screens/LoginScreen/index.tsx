import React from 'react';
import { CompositeScreenProps } from '@react-navigation/native';
import { AuthStackParamList, RootStackParamList } from '@src/types/navigation';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { useLoginMutation } from '@src/hooks/mutations/useLoginMutation';
import { useForm, Controller } from 'react-hook-form';
import { useRecoilValue } from 'recoil';
import { userState } from '@src/recoils';
import { getUniqueId } from 'react-native-device-info';
import Text from '@src/components/Text';
import Button from '@src/components/Button';
import { APP_NAME } from '@env';
import GoogleIcon from '@src/components/GoogleIcon';
import NaverIcon from '@src/components/NaverIcon';
import KakaoIcon from '@src/components/KakaoIcon';
import KeyboardAvoidingScrollView from '@src/components/KeyboardAvoidingScrollView';
import { HeaderHeightContext } from '@react-navigation/elements';
import { useSetUser } from './hooks/useSetUser';
import { useSetErrorMessages } from './hooks/useSetErrorMessages';
import { useSuccess } from './hooks/useSuccess';
import {
  Container,
  TitleContainer,
  Greeting,
  Submit,
  SNSTitleContainer,
  Line,
  SNSTitle,
  SNSButtonsContainer,
  SNSButton,
  LinksContainer,
  Divider,
  ErrorMessage,
} from './styled';
import useRules from './hooks/useRules';
import useRenders from './hooks/useRenders';

type P = CompositeScreenProps<
  NativeStackScreenProps<AuthStackParamList, 'Login'>,
  NativeStackScreenProps<RootStackParamList>
>;

export type LoginFormInput = {
  email: string;
  password: string;
};

const LoginScreen: React.FC<P> = ({ navigation }) => {
  const headerHeight = React.useContext(HeaderHeightContext) || 0;
  const user = useRecoilValue(userState);
  const {
    control,
    handleSubmit,
    formState: { isSubmitting },
  } = useForm<LoginFormInput>();
  const [login, { data, error, loading }] = useLoginMutation();
  const { emailRules, passwordRules } = useRules();
  const [errorMessages, setErrorMessages] = React.useState<string[]>([]);
  const onSubmit = async (input: LoginFormInput) => {
    await login({
      variables: {
        input: {
          ...input,
          device_id: getUniqueId(),
        },
      },
    });
  };
  const { emailRender, passwordRender } = useRenders(handleSubmit(onSubmit));
  const onGoogleLogin = () => {};
  const onAppleLogin = () => {};
  const onNaverLogin = () => {};
  const onKakaoLogin = () => {};
  const onRegister = () => navigation.navigate('Register');
  const onFindPassword = () => {};

  useSetUser(data);
  useSetErrorMessages(setErrorMessages, error);
  useSuccess(navigation, user);

  return (
    <KeyboardAvoidingScrollView>
      <Container headerHeight={headerHeight}>
        <TitleContainer>
          <Text size={40} weight="bold">
            üí™ {APP_NAME}
          </Text>
          <Greeting size={22}>Î∞òÍ∞ÄÏõåÏöî!</Greeting>
          <Text size={22} weight="bold">
            Í∑ºÏú°Îß®ÏúºÎ°ú ÎìùÍ∑ºÌïòÏÑ∏Ïöî
          </Text>
        </TitleContainer>

        <Controller
          render={emailRender}
          name="email"
          control={control}
          rules={emailRules}
        />
        <Controller
          render={passwordRender}
          name="password"
          control={control}
          rules={passwordRules}
        />
        {errorMessages.map(message => (
          <ErrorMessage key={message} size={12} color="error">
            {message}
          </ErrorMessage>
        ))}
        <Submit
          title="Î°úÍ∑∏Ïù∏"
          onPress={handleSubmit(onSubmit)}
          loading={isSubmitting || loading}
        />

        <SNSTitleContainer>
          <Line />
          <SNSTitle size={13} color="grey3">
            SNSÎ°ú 1Ï¥àÎßåÏóê Î°úÍ∑∏Ïù∏
          </SNSTitle>
          <Line />
        </SNSTitleContainer>
        <SNSButtonsContainer>
          <SNSButton
            title={<GoogleIcon />}
            onPress={onGoogleLogin}
            color="grey5"
          />
          <SNSButton
            icon={{
              name: 'apple',
              type: 'font-awesome-5',
              color: 'background',
            }}
            onPress={onAppleLogin}
            color="foreground"
          />
          <SNSButton
            title={<NaverIcon />}
            onPress={onNaverLogin}
            color="naver"
          />
          <SNSButton
            title={<KakaoIcon />}
            onPress={onKakaoLogin}
            color="kakao"
          />
        </SNSButtonsContainer>

        <LinksContainer>
          <Button
            testID="registerButton"
            onPress={onRegister}
            type="clear"
            title="ÌöåÏõêÍ∞ÄÏûÖ"
            color="grey3"
            weight="normal"
          />
          <Divider color="grey3">|</Divider>
          <Button
            onPress={onFindPassword}
            type="clear"
            title="ÎπÑÎ∞ÄÎ≤àÌò∏Ï∞æÍ∏∞"
            color="grey3"
            weight="normal"
          />
        </LinksContainer>
      </Container>
    </KeyboardAvoidingScrollView>
  );
};

export default LoginScreen;
