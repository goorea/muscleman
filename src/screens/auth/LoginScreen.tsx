import React, { useEffect } from 'react';
import { CompositeScreenProps } from '@react-navigation/native';
import { AuthStackParamList, RootStackParamList } from '@src/types/navigation';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { Button, Input } from 'react-native-elements';
import Loader from '@src/components/Loader';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { flash } from '@src/functions';
import { useLoginMutation } from '@src/hooks/mutations/useLoginMutation';
import { useForm, Controller } from 'react-hook-form';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import { tokenState, userQuery } from '@src/recoils';
import styled from 'styled-components/native';
import { flexFillCenter } from '@src/styles/flex';
import { getUniqueId } from 'react-native-device-info';
import Text from '@src/components/Text';

const Container = styled.View`
  ${flexFillCenter}
`;

type P = CompositeScreenProps<
  NativeStackScreenProps<AuthStackParamList, 'Login'>,
  NativeStackScreenProps<RootStackParamList>
> & {};

type FormData = {
  email: string;
  password: string;
};

const LoginScreen: React.FC<P> = ({ navigation }) => {
  const user = useRecoilValue(userQuery);
  const setToken = useSetRecoilState(tokenState);
  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<FormData>({
    defaultValues: {
      email: 'jhs851@naver.com',
      password: '123123123',
    },
  });
  const [login, { data, loading }] = useLoginMutation();
  const onSubmit = async (input: FormData) => {
    await login({
      variables: { input: { ...input, device_id: getUniqueId() } },
    });
  };

  useEffect(() => {
    if (data) {
      const { token, refresh_token } = data.login;

      AsyncStorage.multiSet([
        ['@token', token],
        ['@refresh_token', refresh_token],
      ]).then(() => setToken(token));
    }
  }, [data, setToken]);

  useEffect(() => {
    if (navigation && user) {
      flash({
        title: `${user.name}님 환영합니다!`,
        contents: '오늘도 득근하세요 !!!',
        type: 'success',
      });
      navigation.goBack();
    }
  }, [navigation, user]);

  if (loading) {
    return <Loader />;
  }

  return (
    <Container>
      <Text>Login Screen</Text>
      <Controller
        render={({ field }) => (
          <Input
            leftIcon={{ name: 'email' }}
            label="이메일"
            errorMessage={errors.email?.message}
            keyboardType="email-address"
            returnKeyType="next"
            onChangeText={field.onChange}
            {...field}
          />
        )}
        name="email"
        control={control}
        rules={{
          required: '이메일을 입력 해주세요',
        }}
      />
      <Controller
        render={({ field }) => (
          <Input
            leftIcon={{ name: 'lock' }}
            label="비밀번호"
            errorMessage={errors.password?.message}
            returnKeyType="done"
            secureTextEntry={true}
            onChangeText={field.onChange}
            {...field}
          />
        )}
        name="password"
        control={control}
        rules={{
          required: '비밀번호를 입력 해주세요',
        }}
      />
      <Button title="Login" onPress={handleSubmit(onSubmit)} />
      <Button
        title="Register"
        onPress={() => navigation.navigate('Register')}
      />
    </Container>
  );
};

export default LoginScreen;
