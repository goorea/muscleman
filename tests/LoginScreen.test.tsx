import 'react-native';
import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import LoginScreen from '@src/screens/auth/LoginScreen';
import { navigationMock, navigationNavigateMock } from '@mocks/navigationMocks';
import { wrapper } from '@tests/functions';

describe('LoginScreen 컴포넌트', () => {
  const rendered = () =>
    render(
      <LoginScreen
        navigation={navigationMock}
        route={{
          key: '',
          name: 'Login',
        }}
      />,
      { wrapper },
    );

  it('렌더링이 올바르게 된다', () => {
    const { toJSON } = rendered();

    expect(toJSON()).toMatchSnapshot();
  });

  it('이메일 필드와 비밀번호 필드와 로그인 버튼이 보인다', () => {
    const { queryByTestId, queryByText } = rendered();

    expect(queryByTestId('emailField')).not.toBeNull();
    expect(queryByTestId('passwordField')).not.toBeNull();
    expect(queryByText('로그인')).not.toBeNull();
  });

  it('이메일 필드와 비밀번호 필드를 비우고 로그인 버튼을 누르면 에러 메세지들이 보인다', async () => {
    const { queryByText, getByText } = rendered();

    expect(queryByText('이메일을 입력 해주세요')).toBeNull();
    expect(queryByText('비밀번호를 입력 해주세요')).toBeNull();

    await waitFor(() => fireEvent.press(getByText('로그인')));

    expect(queryByText('이메일을 입력 해주세요')).not.toBeNull();
    expect(queryByText('비밀번호를 입력 해주세요')).not.toBeNull();
  });

  it('이메일 필드는 이메일 형식이 아니라면 에러 메세지가 보인다', async () => {
    const { getByTestId, queryByText, getByText } = rendered();

    fireEvent.changeText(getByTestId('emailField'), 'Hello World');

    await waitFor(() => fireEvent.press(getByText('로그인')));

    expect(queryByText('이메일 형식이 아닙니다')).not.toBeNull();
  });

  it('비밀번호 필드는 8글자 이상이어야 한다', async () => {
    const { getByTestId, queryByText, getByText } = rendered();

    fireEvent.changeText(getByTestId('emailField'), 'john@example.com');
    fireEvent.changeText(getByTestId('passwordField'), 'a'.repeat(7));

    await waitFor(() => fireEvent.press(getByText('로그인')));

    expect(queryByText('8글자보다 적습니다')).not.toBeNull();
  });

  it('회원가입 페이지로 이동할 수 있다', async () => {
    const { getByTestId } = rendered();

    await waitFor(() => fireEvent.press(getByTestId('register-button')));

    expect(navigationNavigateMock).toHaveBeenCalledWith('Register');
  });

  // TODO: 비밀번호찾기 페이지로 이동할 수 있다
});
