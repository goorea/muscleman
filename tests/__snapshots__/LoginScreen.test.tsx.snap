// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen 컴포넌트 렌더링이 올바르게 된다 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        headerHeight={0}
        style={
          Array [
            Object {
              "height": 1334,
              "paddingBottom": 30,
              "paddingLeft": 30,
              "paddingRight": 30,
              "paddingTop": 42,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <Text
            italic={false}
            size={40}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "SpoqaHanSansNeo-Bold",
                  "fontSize": 40,
                  "fontStyle": "normal",
                },
              ]
            }
            weight="bold"
          >
            💪 
            근육맨
          </Text>
          <Text
            italic={false}
            size={22}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "SpoqaHanSansNeo-Regular",
                  "fontSize": 22,
                  "fontStyle": "normal",
                },
                Object {
                  "marginTop": 16,
                },
              ]
            }
          >
            반가워요!
          </Text>
          <Text
            italic={false}
            size={22}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "SpoqaHanSansNeo-Bold",
                  "fontSize": 22,
                  "fontStyle": "normal",
                },
              ]
            }
            weight="bold"
          >
            근육맨으로 득근하세요
          </Text>
        </View>
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "baseline",
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "marginBottom": 4,
                  "paddingBottom": 10,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <Text
              collapsable={false}
              color="foreground"
              nativeID="animatedComponent"
              onLayout={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SpoqaHanSansNeo-Regular",
                  "fontSize": 16,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 16,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              이메일
            </Text>
            <TextInput
              autoCompleteType="email"
              keyboardType="email-address"
              label="이메일"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "SpoqaHanSansNeo-Regular",
                    "fontSize": 16,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                    "width": "100%",
                  },
                ]
              }
              testID="emailField"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "baseline",
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "marginBottom": 4,
                  "paddingBottom": 10,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 10,
                },
              ]
            }
          >
            <Text
              collapsable={false}
              color="foreground"
              nativeID="animatedComponent"
              onLayout={[Function]}
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SpoqaHanSansNeo-Regular",
                  "fontSize": 16,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 16,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              비밀번호
            </Text>
            <TextInput
              label="비밀번호"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "SpoqaHanSansNeo-Regular",
                    "fontSize": 16,
                    "paddingBottom": 0,
                    "paddingLeft": 0,
                    "paddingRight": 0,
                    "paddingTop": 0,
                    "width": "100%",
                  },
                ]
              }
              testID="passwordField"
            />
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2687ED",
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "paddingBottom": 14,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 14,
            }
          }
        >
          <Text
            color="white"
            italic={false}
            size={16}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "SpoqaHanSansNeo-Bold",
                  "fontSize": 16,
                  "fontStyle": "normal",
                },
              ]
            }
            weight="bold"
          >
            로그인
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 30,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 1,
                },
              ]
            }
          />
          <Text
            color="grey3"
            italic={false}
            size={13}
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontFamily": "SpoqaHanSansNeo-Regular",
                  "fontSize": 13,
                  "fontStyle": "normal",
                },
                Object {
                  "paddingBottom": 0,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 0,
                },
              ]
            }
          >
            SNS로 1초만에 로그인
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e1e8ee",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 1,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginTop": 20,
                "paddingBottom": 0,
                "paddingLeft": 30,
                "paddingRight": 30,
                "paddingTop": 0,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#e1e8ee",
                "borderRadius": 23,
                "height": 46,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 46,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={20}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      27.009001,
                      -39.238998,
                    ]
                  }
                >
                  <RNSVGPath
                    d="M -3.264 51.509 C -3.264 50.719 -3.334 49.969 -3.454 49.239 L -14.754 49.239 L -14.754 53.749 L -8.284 53.749 C -8.574 55.229 -9.424 56.479 -10.684 57.329 L -10.684 60.329 L -6.824 60.329 C -4.564 58.239 -3.264 55.159 -3.264 51.509 Z"
                    fill={4282549748}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M -14.754 63.239 C -11.514 63.239 -8.804 62.159 -6.824 60.329 L -10.684 57.329 C -11.764 58.049 -13.134 58.489 -14.754 58.489 C -17.884 58.489 -20.534 56.379 -21.484 53.529 L -25.464 53.529 L -25.464 56.619 C -23.494 60.539 -19.444 63.239 -14.754 63.239 Z"
                    fill={4281641043}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M -21.484 53.529 C -21.734 52.809 -21.864 52.039 -21.864 51.239 C -21.864 50.439 -21.724 49.669 -21.484 48.949 L -21.484 45.859 L -25.464 45.859 C -26.284 47.479 -26.754 49.299 -26.754 51.239 C -26.754 53.179 -26.284 54.999 -25.464 56.619 L -21.484 53.529 Z"
                    fill={4294687749}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGPath
                    d="M -14.754 43.989 C -12.984 43.989 -11.404 44.599 -10.154 45.789 L -6.734 42.369 C -8.804 40.429 -11.514 39.239 -14.754 39.239 C -19.444 39.239 -23.494 41.939 -25.464 45.859 L -21.484 48.949 C -20.534 46.099 -17.884 43.989 -14.754 43.989 Z"
                    fill={4293542709}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000000",
                "borderRadius": 23,
                "height": 46,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 46,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome5Brands-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
              type="font-awesome-5"
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1ec800",
                "borderRadius": 23,
                "height": 46,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 46,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={18.285714286}
              bbWidth={20}
              focusable={false}
              height={18.285714286}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 18,
                    "width": 20,
                  },
                ]
              }
              vbHeight={25.6}
              vbWidth={28}
              width={20}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M18.5 0 18.5 12.9 9.6 0 0 0 0 25.7 9.6 25.7 9.6 12.7 18.4 25.7 28 25.7 28 0 z"
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fee500",
                "borderRadius": 23,
                "height": 46,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "width": 46,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={25.74444432}
              bbWidth={28}
              focusable={false}
              height={25.74444432}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 25,
                    "width": 28,
                  },
                ]
              }
              vbHeight={33.1}
              vbWidth={36}
              width={28}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M18,0C8.1,0,0,6.4,0,14.2c0,5.1,3.4,9.6,8.5,12.1c-0.4,1.4-1.4,5.1-1.6,5.9c-0.2,1,0.4,1,0.8,0.7c0.3-0.2,4.9-3.3,6.9-4.7c1.1,0.2,2.2,0.2,3.3,0.2c9.9,0,18-6.4,18-14.2S27.9,0,18,0z"
                  fill={4281867807}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M11.2,11.1c-1.3,0-5.7,0-5.7,0c-1.1,0-1.1,1.8,0,1.8h1.9c0,0,0,5.1,0,6.3c0,0.9,2,0.9,1.9,0l0-6.3c0,0,0.6,0,1.9,0C12.4,13,12.4,11.1,11.2,11.1z"
                  fill={4293842977}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M15.6,11.7c-0.3-0.6-1.9-0.9-2.3,0c-0.5,1-2.1,5.8-2.7,7.1c-0.4,0.9,1.3,1.7,1.7,0.8l0.4-1.3l3.4,0c0,0-0.1,0.1,0.3,1.3c0.4,1,2.2,0.3,1.8-0.7C17.8,17.4,16,12.4,15.6,11.7z M13.4,16.5l1.1-3.3l1,3.3H13.4z"
                  fill={4293842977}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M23.5,18c-0.9,0-2.5,0-2.5,0s0-4.9,0-6c0-1.1-2-1.1-2,0c0,1.6,0,6.9,0,6.9s-0.2,1,0.7,1c0.9,0,2.9,0,3.8,0S24.4,18,23.5,18z"
                  fill={4293842977}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  d="M31.1,18.6l-2.6-3.5c0,0,1.8-1.9,2.5-2.6c0.4-0.4-0.9-1.7-1.3-1.3c-0.4,0.4-3.2,3.2-3.2,3.2s0-1.8,0-2.7c0-0.8-1.9-0.7-1.9,0.1c0,0.7,0,7.2,0,7.7c0,0.8,1.9,0.8,1.9,0c0-0.9,0-2.5,0-2.5l0.6-0.5l2.5,3.4C30.1,20.4,31.6,19.2,31.1,18.6z"
                  fill={4293842977}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": "auto",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              }
            }
            testID="registerButton"
          >
            <Text
              color="grey3"
              italic={false}
              size={16}
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontFamily": "SpoqaHanSansNeo-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                  },
                ]
              }
              weight="normal"
            >
              회원가입
            </Text>
          </View>
          <Text
            color="grey3"
            italic={false}
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontFamily": "SpoqaHanSansNeo-Regular",
                  "fontSize": 16,
                  "fontStyle": "normal",
                },
                Object {
                  "marginBottom": 0,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 0,
                },
              ]
            }
          >
            |
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              }
            }
          >
            <Text
              color="grey3"
              italic={false}
              size={16}
              style={
                Array [
                  Object {
                    "color": "#86939e",
                    "fontFamily": "SpoqaHanSansNeo-Regular",
                    "fontSize": 16,
                    "fontStyle": "normal",
                  },
                ]
              }
              weight="normal"
            >
              비밀번호찾기
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
